openprescribing/frontend/views/views.py

```diff
@@ -878,6 +889,19 @@ def all_england_price_per_unit_by_presentation(request, bnf_code):
     return render(request, "price_per_unit.html", context)
 
 
+def _get_generic_bnf_code(bnf_code):
```

Comment:
Does this have to use substitution sets?  I think you can do it just by transforming the BNF code.

My understanding is that given a branded BNF code that ends `YZ`, you could get the generic equivalent by replacing the last six characters with `AAYZYZ`.

Reply:
That doesn't account for formulation swaps though. And in general, I'd rather have that logic centralised as far as possible.

Reply:
Ah right.  In which case the name of the function is slightly misleading, as it's doing more than getting a generic BNF code.

Reply:
Yeah, we don't really have a good term for this yet. I originally called it "get_canonical_bnf_code" (because I was thinking in terms of the URL) but that isn't quite right so I changed it to "generic". Open to suggestions.

Reply:
`get_primary_substitutable_bnf_code` is a bit of a mouthful, but it's less confusing than `get_generic_bnf_code`.